clients/src/main/java/org/apache/kafka/common/record/Record.java

```
@@ -100,149 +103,17 @@ public Record(ByteBuffer buffer) {
     }
 
     // Package private constructor for inner iteration.
```

Update the comment as well for `public`. Also I'm wondering if we could rename `wrapperXX` just to `XX` and add comments indicating that they are only used for old formatted messages with magic number > 0, and also add a check in constructor that the `magic()` field is consistent with its values: if it is larger than 0 these two fields should never be null; if it is 0 then these two fields should always be null etc.

---

The wrapper values can be null if either the magic is 0 or if the record is uncompressed. I'll add an assertion for this, but I'd prefer to keep the current names since it's otherwise harder to explain.

---

Actually I had to remove these assertions. First, if the magic is 1, then we can't distinguish between a shallow uncompressed entry, and a deep entry, which would also not have its compression flag enabled. We could at least raise an error if magic is 0 and there is a provided wrapper timestamp., but we currently have test cases which allow us to create a Record with only a valid CRC (which means we can't check magic). We could probably be a little stricter: if there aren't enough bytes in the ByteBuffer to read the record overhead, then obviously the record can't be valid, so maybe those test cases are kind of silly and should be removed.